pkg/trace/api/evp_intake_proxy.go

```diff
@@ -0,0 +1,252 @@
+		msg := fmt.Sprintf("EVPProxy is disabled: %v", message)
+		http.Error(w, msg, http.StatusMethodNotAllowed)
+	})
+}
+
+// evpProxyForwarder creates an http.ReverseProxy which can forward payloads to
+// one or more endpoints, based on the request received and the Agent configuration.
+// Headers are not proxied, instead we add our own known set of headers.
+// See also evpProxyTransport below.
+func evpProxyForwarder(conf *config.AgentConfig, endpoints []config.Endpoint, transport http.RoundTripper, logger *stdlog.Logger) http.Handler {
+	director := func(req *http.Request) {
+		if req == nil {
+			return
+		}
+
+		containerID := req.Header.Get("Datadog-Container-ID")
+		contentType := req.Header.Get("Content-Type")
+		userAgent := req.Header.Get("User-Agent")
+
+		// Clear all received headers
+		req.Header = http.Header{}
+
+		// Standard headers
+		req.Header.Set("Via", fmt.Sprintf("trace-agent %s", info.Version))
+		if contentType != "" {
+			req.Header.Set("Content-Type", contentType)
+		}
+		req.Header.Set("User-Agent", userAgent) // Set even if an empty string so Go doesn't set its default
+		req.Header["X-Forwarded-For"] = nil     // Prevent setting X-Forwarded-For
+
+		// Datadog headers
+		if ctags := getContainerTags(conf.ContainerTags, containerID); ctags != "" {
+			req.Header.Set("X-Datadog-Container-Tags", ctags)
+		}
+		req.Header.Set("X-Datadog-Hostname", conf.Hostname)
+		req.Header.Set("X-Datadog-AgentDefaultEnv", conf.DefaultEnv)
+
+		// URL, Host and the API key header are set in the transport for each outbound request
+	}
+
+	return &httputil.ReverseProxy{
+		Director:  director,
+		ErrorLog:  logger,
+		Transport: &evpProxyTransport{transport, endpoints, conf.EVPProxy.MaxPayloadSize},
+	}
+}
+
+// evpProxyTransport sends HTTPS requests to multiple targets using an
+// underlying http.RoundTripper. API keys are set separately for each target.
+// When multiple endpoints are in use the response from the first endpoint
+// is proxied back to the client, while for all aditional endpoints the
+// response is discarded.
+type evpProxyTransport struct {
+	transport      http.RoundTripper
+	endpoints      []config.Endpoint
+	maxPayloadSize int64
+}
+
+func (t *evpProxyTransport) RoundTrip(req *http.Request) (rresp *http.Response, rerr error) {
+
+	// Metrics with stats for debugging
+	beginTime := time.Now()
+	metricTags := []string{}
+	var contentLength int64
+	defer func() {
+		metrics.Count("datadog.trace_agent.evp_proxy.request", 1, metricTags, 1)
+		metrics.Count("datadog.trace_agent.evp_proxy.request_bytes", contentLength, metricTags, 1)
+		metrics.Timing("datadog.trace_agent.evp_proxy.request_duration_ms", time.Since(beginTime), metricTags, 1)
+		if rerr != nil {
+			metrics.Count("datadog.trace_agent.evp_proxy.request_error", 1, metricTags, 1)
+		}
+	}()
+
+	if req == nil || req.URL == nil {
+		return nil, fmt.Errorf("EVPProxy: invalid request")
+	}
+
```

Comment:
With the risk of being annoying here, can you please cut down on empty lines? We're trying to follow the guidelines from our team in this repo: https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines#general-guidelines (see point 3)